coding-guides/using-audit-logs.md

```diff
@@ -0,0 +1,169 @@
+  executor: [Object],
+  changes: null,
+  id: '434116792567201792',
+  extra: [Object],
+  target: [Object] }
+```
+Notice the `reason` field. Some audit logs, like kicking and banning, can provide a reason. You can probably make the bot log when a user is banned and for whatever reason. What we want is the executor of the action. We do that by going to the `executor` target as that is where the user is stored. Below is showing what the information that the `executor` property provides us:
+```js
+User {
+  id: '286270602820452353',
+  username: 'Zoth The Wumpus',
+  discriminator: '6066',
+  avatar: '57361ef0f8e23e02a44069c3dd5f5f47',
+  bot: false,
+  lastMessageID: '435165896198062091',
+  lastMessage: [Object] }
+```
+
+With all the information above, we can start creating comparisons to narrow down on who really deleted the message, whether it was the author of the message or someone else.
+
+```js
+  // Please keep in mind: Discord's audit logs will not log the information if the author of that message deleted it.
+  // I did this with a series of checks. 
+  // We are checking if the channel ID is the same as the id the message is deleted it
+  if ((await message.guild.fetchAuditLogs({type: 'MESSAGE_DELETE'}).then(g => g.entries.first().extra.channel.id === message.channel.id)) 
+  
+  //Then we are checking if the target is the same as the author id
+  && (await message.guild.fetchAuditLogs({type: 'MESSAGE_DELETE'}).then(g => g.entries.first().target.id === message.author.id)) 
+  
+  // We are comparing time as audit logs are sometimes slow. 
+  && (await message.guild.fetchAuditLogs({type: 'MESSAGE_DELETE'}).then(g => g.entries.first().createdTimestamp > (Date.now() - 5000)))) {
+    user = executor
+  } else if ((await message.guild.fetchAuditLogs({type: 'MESSAGE_DELETE'}).then(g => g.entries.first().extra.channel.id === message.channel.id)) 
+  && (await message.guild.fetchAuditLogs({type: 'MESSAGE_DELETE'}).then(g => g.entries.first().target.id === message.author.id)) 
+  && (await message.guild.fetchAuditLogs({type: 'MESSAGE_DELETE'}).then(g => g.entries.first().createdTimestamp > (Date.now() - 5000)))
+  
+  // Everything is the same as above, however, this section. This part is telling us if the count is greater than 1.
+  // If it is, then the executor must have deleted it.
+  && (await message.guild.fetchAuditLogs({type: 'MESSAGE_DELETE'}).then(g => g.entries.first().extra.count > 1))) {
+    user = executor
+ } else { 
+    
+    // When all else fails, we can assume that the deleter is the author.
+    user = message.author.username
+  }
+```
+
+With all the given information above, let's start sending it all to a channel.
+
+```js
+  // I always format my messages in strings as it's easier (for me) to understand what I am doing as some strings can get pretty long,
+  // like the $5 foot long from subway. Man I miss that. 
+  const deletedMessageInformation = `A message was deleted in ${message.channel.name} by ${user}`;
```

Comment:
Creating a variable for exactly one use is a waste of cycles. 

Do this instead:
```js
logs.send(`A message was deleted in ${message.channel.name} by ${user}`);
```